src/binary/ibc-setup/commands/keys-list.ts

```diff
@@ -0,0 +1,93 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import readline from 'readline';
+
+import { registryFile } from '../../constants';
+import { deriveAddress } from '../utils/derive-address';
+import { getDefaultHomePath } from '../utils/get-default-home-path';
+import { loadAndValidateRegistry } from '../utils/load-and-validate-registry';
+import { resolveRequiredOption } from '../utils/resolve-required-option';
+
+type Flags = {
+  readonly interactive: boolean;
+  readonly home?: string;
+  readonly keyFile?: string;
+  readonly mnemonic?: string;
+};
+
+export type Options = {
+  readonly home: string;
+  readonly mnemonic: string;
+};
+
+export async function readMnemonicFromStdin(interactive: Flags['interactive']) {
+  if (!interactive) {
+    return undefined;
+  }
+
+  const readlineInterface = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  const mnemonic = await new Promise<string>((resolve) => {
+    readlineInterface.question('enter mnemonic phrase: ', (stdin) => {
+      readlineInterface.close();
+      resolve(stdin);
+    });
+  });
+
+  return mnemonic;
+}
+
+function readMnemonicFromFile(keyFile: Flags['keyFile']) {
```

Comment:
I would pull both these key file functions into utils. We need them as well when listing balances as well. Maybe add some flag parsing/handling logic into that utils as well, as it is the same for all commands that use it (the second command to use it is in the same issue)

Reply:
Sure, this will be reused between commands.